dhcpv4/bsdp/bsdp.go

```diff
@@ -51,6 +51,29 @@ func needsReplyPort(replyPort uint16) bool {
 	return replyPort != 0 && replyPort != dhcpv4.ClientPort
 }
 
+// MessageTypeFromPacket extracts the BSDP message type (LIST, SELECT) from the
+// vendor-specific options and returns it. If the message type option cannot be
+// found, returns false.
+func MessageTypeFromPacket(packet *dhcpv4.DHCPv4) (MessageType, bool) {
```

Comment:
I think this function could have returned just a `*MessageType` instead of `(MessageType, bool)`. If the return value is nil, it signals that the function could not find the message type.
Adding a second return value (the boolean) makes the function more verbose (e.g. the return value cannot be passed directly to another function but requires a check on the boolean), and the object is passed by copy in any case, either that it succeeds or fails

Reply:
Good point, fixed with #162